src/argilla_sdk/records/_io/_generic.py

```diff
@@ -35,10 +36,10 @@ def to_dict(self, flatten: bool = False, orient: str = "names") -> Dict[str, Any
         Returns:
             dataset_records (Dict[str, Union[str, float, int, list]]): The exported records in a dictionary format.
         """
-        records = [r for r in self]
-        return self._export_to_dict(records=records, flatten=flatten, orient=orient)
+        return GenericIO._export_to_dict(records=records, flatten=flatten, orient=orient)
```

Comment:
since we have a one-liner method, maybe is more readable to move the logic inside here and remove the private one.